File: src/main/java/com/wesker926/leetcode/algorithms/p0299/s1/Solution.java
package com.wesker926.leetcode.algorithms.p0299.s1;

/**
 * @author wesker.gh
 * @date 2021/11/8
 * @description 哈希表
 */
public class Solution {
    public String getHint(String secret, String guess) {
        int[] set = new int[10];
        int match = 0, same = 0;
        char[] a = secret.toCharArray(), b = guess.toCharArray();
        for (int i = 0; i < a.length; i++) {
            set[a[i] - '0']++;
            if (a[i] == b[i]) {
                match++;
            }
        }
        for (char c : b) {
            if (set[c - '0'] > 0) {
                set[c - '0']--;
                same++;
            }
        }
        same -= match;
        return new StringBuilder().append(match).append('A').append(same).append('B').toString();
    }
}

File: src/main/java/com/wesker926/leetcode/algorithms/p0299/s1/solution.go
// Package s1
// @author: wesker.gh
// @date: 2021/11/8
// @description: 哈希表
package s1

import "strconv"

func getHint(secret string, guess string) string {
	set, match, same := [10]int{}, 0, 0
	for i := 0; i < len(secret); i++ {
		set[secret[i]-'0']++
		if secret[i] == guess[i] {
			match++
		}
	}
	for i := 0; i < len(guess); i++ {
		if set[guess[i]-'0'] > 0 {
			set[guess[i]-'0']--
			same++
		}
	}
	same -= match
	return strconv.Itoa(match) + "A" + strconv.Itoa(same) + "B"
}
